src/opm/output/eclipse/AggregateNetworkData.cpp

```diff
@@ -0,0 +1,752 @@
+        ind = (dwntr_nd_res.first) ? dwntr_nd_res.second + 1 : 0 ;
+        inlets.push_back(ind);
+        auto uptr_nd_res = findInVector<std::string>(ntwNdNm, branch->uptree_node());
+        ind = (dwntr_nd_res.first) ? uptr_nd_res.second + 1 : 0 ;
+        outlets.push_back(ind);
+    }
+
+    int n1 = inlets[0];
+    int ind_br = 0;
+    newInobr.push_back(n1 * (-1));
+    inlets[0] = used_flag;
+
+    while (static_cast<std::size_t>(n1) <= ntwNdNm.size()) {
+        ind_br = next_branch(n1, inlets, outlets);
+        while (ind_br != 0) {
+            newInobr.push_back(ind_br);
+            if (ind_br > 0) {
+                outlets[ind_br-1] = used_flag;
+            } else {
+                inlets[ind_br*(-1) - 1] = used_flag;
+            }
+            ind_br = next_branch(n1, inlets, outlets);
+        }
+        n1 += 1;
+    }
+    return newInobr;
+}
+
+bool fixedPressureNode(const Opm::Schedule&         sched,
+                       const std::string&           nodeName,
+                       const size_t                 lookup_step
+                      )
+{
+    auto& network = sched[lookup_step].network();
+    bool fpn = (network.node(nodeName).terminal_pressure().has_value()) ? true  : false;
+    return fpn;
+}
+
+double nodePressure(const Opm::Schedule&               sched,
+                    const ::Opm::SummaryState&         smry,
+                    const std::string&                 nodeName,
+                    const Opm::UnitSystem&             units,
+                    const size_t                       lookup_step
+                   )
+{
+    using M = ::Opm::UnitSystem::measure;
+    double node_pres = 1.;
+    bool node_wgroup = false;
+    const auto& wells = sched.getWells(lookup_step);
+    auto& network = sched[lookup_step].network();
+
+    // If a node is a well group, set the node pressure to the well's thp-limit if this is larger than the default value (1.)
+    for (const auto& well : wells) {
+        const auto& wgroup_name = well.groupName();
+        if (wgroup_name == nodeName) {
+            if (well.isProducer()) {
+                const auto& pc = well.productionControls(smry);
+                if (pc.thp_limit >= node_pres) {
+                    node_pres = units.from_si(M::pressure, pc.thp_limit);
+                    node_wgroup = true;
+                }
+            }
+        }
+    }
+
+    // for nodes that are not well groups, set the node pressure to the fixed pressure potentially higher in the node tree
+    if (!node_wgroup) {
+        if (fixedPressureNode(sched, nodeName, lookup_step)) {
+            // node is a fixed pressure node
+            node_pres = units.from_si(M::pressure, network.node(nodeName).terminal_pressure().value());
+        }
+        else {
+            // find fixed pressure higher in the node tree
+            bool fp_flag = false;
+            auto node_name = nodeName;
```

Comment:
What is the point of variable `node_name = nodeName`?

Reply:
The variable node_name may in some cases change to the name of an uptree node with fixed pressure. The nodeName is a const.